iris_ugrid/ugrid_cf_reader.py

```diff
@@ -218,16 +231,34 @@ def cube_completion_adjust(self, cube):
                 )
                 node_coordinates.append(name)
 
-            cube.ugrid = CubeUgrid(
+            cube_ugrid = CubeUgrid(
                 cube_dim=i_dim,
                 grid=mesh,
                 mesh_location=mesh_location,
                 topology_dimension=topology_dimension,
                 node_coordinates=sorted(node_coordinates),
             )
-        else:
-            # Add an empty 'cube.ugrid' to all cubes otherwise.
-            cube.ugrid = None
+            # Return a new UCube, based on the provided Cube, and replacing it
+            # in the caller (and as returned to user).
+            # Absolutely **everything** is the same, except for the extra ugrid
+            # property.
+            new_result_cube = UCube(
+                data=cube.core_data(),
+                standard_name=cube.standard_name,
+                long_name=cube.long_name,
+                var_name=cube.var_name,
+                units=cube.units,
+                attributes=cube.attributes,
+                cell_methods=cube.cell_methods,
+                dim_coords_and_dims=cube._dim_coords_and_dims,
+                aux_coords_and_dims=cube._aux_coords_and_dims,
+                aux_factories=cube.aux_factories,
+                cell_measures_and_dims=cube._cell_measures_and_dims,
+                ancillary_variables_and_dims=cube._ancillary_variables_and_dims,
+                ugrid=cube_ugrid,
+            )
+
```

Comment:
Since this is intended mainly for use within other downstream functions (not even sure why it's public), have decided it would be inappropriate to print a message from this 'level'. @lbdreyer do you agree?

Reply:
Yes I would agree, I don't think it needs to print a message here.